runtime/tests/js/webapis_tests.js

```diff
@@ -1,39 +1,72 @@
-import { assertEquals } from "https://deno.land/std@0.177.0/testing/asserts.ts";
+import { assert, assertEquals } from "https://deno.land/std@0.177.0/testing/asserts.ts";
 
-test("AbortController", () => {
+await test("AbortController", () => {
   assertEquals(typeof AbortController, "function", "typeof AbortController");
   assertEquals(AbortController.name, "AbortController", "AbortController.name");
 });
 
-test("atob & btoa", () => {
+await test("atob & btoa", () => {
   assertEquals(btoa("some text"), "c29tZSB0ZXh0", `btoa("some text)`);
   assertEquals(atob("c29tZSB0ZXh0"), "some text", `atob("c29tZSB0ZXh0")`);
 });
 
-test("TextEncoder", () => {
+await test("fetch", async () => {
+  const res = await fetch("https://google.com/");
+  assertEquals(res.status, 200);
+  const text = await res.text();
+  assertEquals(typeof text, "string");
+  assert(text.includes("<body"));
+});
+
+await test("FormData", async () => {
+  const formData = new FormData();
+  formData.append("name", "value");
+});
+
+await test("Headers", async () => {
+  const headers = new Headers();
+  headers.append("name", "value");
+});
+
+await test("ProgressEvent", async () => {
+  const event = new ProgressEvent();
+  assertEquals(event.total, 0);
+});
+
+await test("Request", async () => {
+  const request = new Request("https://example.com/");
+  await request.arrayBuffer();
+});
+
+await test("Response", async () => {
+  const response = new Response();
+  await response.arrayBuffer();
+});
+
+await test("TextEncoder", () => {
```

Comment:
Oh. Would you mind moving fetch tests into a new file ( `runtime/tests/js/fetch_tests.js`)?

I'd like to start structuring our tests early and keep our test files relatively small.

As a nice side effect, you won't need to change existing tests in this file; they can stay synchronous. You will introduce async tests only in your new test file.

Reply:
Done in 9e43a95